core/src/main/scala/akka/kafka/scaladsl/Committer.scala

```diff
@@ -26,18 +28,48 @@ object Committer {
    */
   def batchFlow(settings: CommitterSettings): Flow[Committable, CommittableOffsetBatch, NotUsed] =
     Flow[Committable]
-    // Not very efficient, ideally we should merge offsets instead of grouping them
       .groupedWeightedWithin(settings.maxBatch, settings.maxInterval)(_.batchSize)
       .map(CommittableOffsetBatch.apply)
       .mapAsync(settings.parallelism) { b =>
         b.commitScaladsl().map(_ => b)(ExecutionContexts.sameThreadExecutionContext)
       }
 
+  /**
+   * API MAY CHANGE
+   *
+   * Batches offsets from context and commits them to Kafka, emits no useful value, but keeps the committed
+   * `CommittableOffsetBatch` as context.
+   */
+  @ApiMayChange
+  def flowWithOffsetContext[E](
+      settings: CommitterSettings
+  ): FlowWithContext[E, Committable, NotUsed, CommittableOffsetBatch, NotUsed] = {
+    val value: Flow[(E, Committable), (NotUsed, CommittableOffsetBatch), NotUsed] = Flow[(E, Committable)]
+      .map(_._2)
+      .groupedWeightedWithin(settings.maxBatch, settings.maxInterval)(_.batchSize)
+      .map(CommittableOffsetBatch.apply)
+      .mapAsync(settings.parallelism) { b =>
+        b.commitScaladsl().map(_ => (NotUsed, b))(ExecutionContexts.sameThreadExecutionContext)
+      }
```

Comment:
Instead of repeating the implementation of `batchFlow`, we could use it here:

```scala
  ...
  .map(_._2)
  .via(batchFlow(settings))
  .map(b => (NotUsed, b))
```